File: Apps/Codaxy.Dextop.Showcase/Demos/Forms/LoginFormWindow.js
Ext.define('Showcase.demos.LoginFormWindow', {
	extend: 'Dextop.Window',
	width: 250,
	autoHeight: true,
	title: 'Please Login',

	initComponent: function () {

		Ext.apply(this, {
			layout: 'fit',
			items: {
				xtype: 'dextopform',
				bodyStyle: 'padding: 5px',
				border: false,
				autoHeight: true,
				model: this.getNestedTypeName('.form.Login'),
				fieldDefaults: {
					labelWidth: 70
				},
				plugins: new Ext.ux.KeyMapPlugin(),
				keys: [{
					key: Ext.EventObject.ENTER,
					scope: this,
					handler: this.doLogin
				}]
			},
			buttons: [{
				text: 'Login',
				formBind: true,
				scope: this,
				handler: this.doLogin
			}]
		});

		this.callParent(arguments);
	},

	doLogin: function () {
		var form = this.down('form').getForm();
		if (form.isValid())
			this.remote.DoLogin(form.getFieldValues(), {
				type: 'alert',
				scope: this,
				prepare: function () { this.setLoading(true); },
				cleanup: function () { this.setLoading(false); },
				success: function (result) {
					this.close();
					alert(result);
				}
			});
	}
});

File: Apps/Codaxy.Dextop.Showcase/Demos/Forms/LoginFormWindow.cs
using System;
using Codaxy.Dextop.Forms;
using Codaxy.Dextop.Remoting;

namespace Codaxy.Dextop.Showcase.Demos.Forms
{
	[Demo("LoginFormWindow",
		Title = "Login Form",
		Description = "Ready to use Login window.",
		Path = "~/Demos/Forms"
	)]
	[LevelBeginner]
	[TopicDextopForms]
	[CategoryCodeSnippet]
	public class LoginFormWindow : DextopWindow
	{
		[DextopRemotable]
		string DoLogin(Login login)
		{
			return "Hello " + login.Username + "!";
		}

		[DextopForm]
		class Login
		{
			[DextopFormField(allowBlank = false, anchor = "0")]
			public String Username { get; set; }

			[DextopFormField(inputType = "password", anchor = "0")]
			public String Password { get; set; }
		}
	}
}